src/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -209,64 +220,54 @@ private unsafe int IndexOfCharArray(char[] anyOf, int startIndex, int count)
         private static unsafe void InitializeProbabilisticMap(uint* charMap, char[] anyOf)
         {
             bool hasAscii = false;
+            uint* charMapLocal = charMap; // https://github.com/dotnet/coreclr/issues/14264
 
             for (int i = 0; i < anyOf.Length; ++i)
             {
-                uint hi, lo;
-                uint c = anyOf[i];
-                lo = c & 0xFF;
-                hi = (c >> 8) & 0xFF;
+                int c = anyOf[i];
+
+                // Map low bit
+                SetCharBit(charMapLocal, (byte)c);
```

Comment:
Note that casting to `byte` (and `SetCharBit` having a `byte`) parameter isn't really necessary. The result of the cast is used in 2 places and neither care about the upper 24 bits. One place is `value & 7`, this is obvious. The other is `(1u << (value >> 3)))`. This is not so obvious - the C# spec says that the shift count is ANDed with 31 and this means that only bits 3-7 of `value` are actually used.

That said, I don't expect removing the cast to improve performance because it's very cheap. So it's probably best to leave it in for clarity.

Reply:
That is some keen thinking and yes the difference is within test variation. Agreed on the clarity point.